File: Python3/number_of_boomerangs.py
class Solution:
    def numberOfBoomerangs(self, points: List[List[int]]) -> int:
        result = 0
        
        for a in points:
            dis_map = {}
            for b in points:
                x = a[0] - b[0] 
                y = a[1] - b[1]
                if (x * x + y * y) not in dis_map: dis_map[x * x + y * y] = 1
                else: dis_map[x * x + y * y] += 1
            for i, index in enumerate(dis_map):
                result += dis_map[index] * (dis_map[index] - 1)
            
        return result

File: C++/number_of_boomerangs.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int result = 0;
        
        for(int i = 0; i < points.size(); i++) {
            map<int, int> dis_map;
            for(int j = 0; j < points.size(); j++) {
                int x = points[i][0] - points[j][0];
                int y = points[i][1] - points[j][1];
                dis_map[x * x + y * y] ++;
            }
            for(auto it = dis_map.begin(); it != dis_map.end(); it++)
                result += it->second * (it->second - 1);
        } 
        
        return result;
    }
};
